main.cpp

```diff
@@ -18,13 +18,27 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
-public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    Rewriter& rewrite;
+ public:
+    CastCallBack(Rewriter& rewriter) rewrite(rewriter) {};
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        const auto *expr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        if (expr != nullptr) {
+            auto loc = CharSourceRange::getCharRange(
+                expr->getLParenLoc(), expr->getSubExprAsWritten()->getBeginLoc());
+            auto tSource = Lexer::getSourceText(
+                CharSourceRange::getTokenRange(
+                    expr->getLParenLoc().getLocWithOffset(1),
+                    expr->getRParenLoc().getLocWithOffset(-1)),
+                    *Result.SourceManager, Result.Context->getLangOpts());
+            auto result = ("static_cast<" + tSource + ">(").str();
+            rewrite.ReplaceText(loc, res);
+            auto closePL = Lexer::getLocForEndOfToken(
+                expr->getSubExprAsWritten()->IgnoreImpCasts()->getEndLoc(), 0,

```

Comment:
Please describe why you are using here `getSubExprAsWritten()` and `IgnoreImpCasts()`.

Reply:
Function `getSubExprAsWritten()` will return the cast subexpression in its original form, unaffected by any implicit casts, as it is in the source code. `IgnoreImpCasts()` then is used to ignore all implicit casts around this expression.

Reply:
The answer is incorrect. Please dump AST to see the difference.

Reply:
Well, it doesn't really matter if we use `getSubExprAsWritten()` along with `IgnoreImpCasts()` in our case, so here the usage of the latter is quite redundant. But nevertheless, I used both functions here to make sure that there are no implicit casts left.